### PR TITLE
Fix typos and en_US/en_GB inconsistency in various files

### PR DESCRIPTION
**Pull request by Abdou-31**

### CODE REVIEW

**Review by TurkeyMcMac — COMMENTED** [review]
Looks good, but I'm not sure about changing the external libraries in the `lib` directory. IMO our changes to these libraries should be kept to a minimum.

**Comment by fluxionary** [comment]
as an american who regularly reads british material (and othe written dialects of english), and as a descriptive linguist, i'm less concerned w/ consistency of spelling and orthography, than in how understandable it is to the target audience. 

contrariwise, i'm certainly a fan of keeping coding style fixed within a single project. that has an analogous import to consistent orthography in natural language, but code style has semantic implications that extend far beyond orthographic variations of english. natural language thrives on ambiguity and variation; under the same condition, code dies a terrible death.

tl/dr: i don't think en_US/en_GB inconsistencies are a bug, and i think spelling variations are acceptable

**Comment by sfan5** [comment]
Stuff in `lib` should indeed not be changed, changes would be overwritten if the libraries are updated (with the exception of Lua maybe).
I'm neutral on the spelling variations, I don't think it's worth spending time on but if someone wants to merge this it's fine.

**Comment by Zughy** [comment]
Thinking about it, I'm actually against such PRs. These inconsistencies show that there are different people from different parts of the world contributing to Minetest (which is great!), so every English variation should be respected. Of course we can't expand this topic to every language, because it'd be quite hard to understand comments in German, Chinese, French etc. altogether, but, since English is used as a bridge between the vast majority of international projects (and for sure it is on Minetest), as long as we're talking about its variations I don't see the problem

**Comment by Abdou-31** [comment]
So? What should I do now? O.O

**Comment by Abdou-31** [comment]
I've restored the files under `lib` directory as recommended.

**Review by TurkeyMcMac — APPROVED** [review]

**Comment by Abdou-31** [comment]
Why did that test fail?

**Comment by TurkeyMcMac** [comment]
It seems unrelated.

**Comment by Abdou-31** [comment]
> 

Ok, but the original commit did pass all the tests successfully...

**Comment by TurkeyMcMac** [comment]
The tests sometimes fail randomly. The failure could be due to a difference in something downloaded during the compilation procedure.

**Comment by Abdou-31** [comment]
> The tests sometimes fail randomly. The failure could be due to a difference in something downloaded during the compilation procedure.

Ok. Thx a lot!

**Comment by Zughy** [comment]
> So? What should I do now? O.O

Not a core dev, but as someone who studies languages and cultures, imho only typos should be fixed due to what I stated earlier. US English vs British English vs Austrlian etc shouldn't be touched, it's not like we're using slangs or jargon. English is English, it's not like USA have an exclusive

**Comment by TurkeyMcMac** [comment]
The spelling doesn't matter much, but American English is specified in the [style guide](https://dev.minetest.net/Code_style_guidelines#Spelling).

**Comment by appgurueu** [comment]
Humans may be tolerant when it comes to recognizing variations, but computers aren't. For comments or docs it is mostly irrelevant, although consistency on spelling is still preferable. But especially for identifiers - variable names etc. - consistent spelling is key to a good developer experience. I don't want to get a compile error while working on a PR because I spelled it `neighbor` instead of `neighbour`. I want to be able to grep for `neighbor` and find what I searched for, both in comments and in code.

**Review by x2048 — APPROVED** [review]
Looks good

**Comment by TurkeyMcMac** [comment]
Rebase needed.

**Comment by Abdou-31** [comment]
> Rebase needed.

Ok. Thx for reviewing. I'll try to rebase qand push asap.

**Comment by Abdou-31** [comment]
Done :+1:

**Comment by Abdou-31** [comment]
I've just messed with those blank lines in `src/serverenvironment.cpp` when rebasing, so the last commit is to fixing that.
